### PR TITLE
Ft get results 166814661

### PR DESCRIPTION
**Pull request by sulenchy**
#### What does this PR do?

- Implement get result functionalities

#### Description of Task to be completed?

- add tests
- add getAllResult method in the rrsult controller
- add /api/v1/result/toplevel route to teachers privilege
- add userRouter
- add test
- add getUserResult method

#### How should this be manually tested?

- Clone the repo
- Run `npm i`
- Run `git pull`
- Checkout to the branch `ft-get-results-166814661`
- Run `npm test`

#### Any background context you want to provide?

- N/A

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -17,8 +18,8 @@ class UsersController {
  * @returns {object} - returns user
  */
   static async signUp(req, res) {
-    const { email, first_name, last_name } = req.body;
-    let { password } = req.body;
+    const { email, first_name, last_name } = toLowerCase(req.body);
+    let { password } = toLowerCase(req.body);
```
Comment (jamsgra) [inline server/controllers/userController.js]: Hey is there a reason to make the password case insensitive?

File: server/middlewares/addResultsValidation.js [hunk 2]
```diff
@@ -40,8 +41,11 @@ class addResultsValidation {
 
       if (Object.keys(fileDuplicates).length) return sendError(res, 422, ExcelValidators.fileDuplicateMessage(fileDuplicates));
 
+      // converts all data to lower case
+      results = Object.values(toLowerCase(results));
```
Comment (jamsgra) [inline server/middlewares/addResultsValidation.js]: I think this should come before `line 31` as we need to make the data lowercase before looking up `emailToUid` in the `Users` table.

We are storing emails in the table as lowercase but a user uploading may capitalize them.

File: server/controllers/userController.js (hunk 1 above)
Comment (sulenchy) [reply server/controllers/userController.js]: That was a mistake on my end. Thanks for pointing out this.

File: server/middlewares/addResultsValidation.js (hunk 2 above)
Comment (sulenchy) [reply server/middlewares/addResultsValidation.js]: Good catch @jamsgra. Thanks man

File: server/controllers/resultsController.js [hunk 3]
```diff
@@ -2,46 +2,161 @@ import models from '../models';
...
+    try {
+      const term = toLowerCase(req.query.term);
+      const year = toLowerCase(req.query.year);
+      const subject = toLowerCase(req.query.subject);
+
+      // specifies options in the findAll sequelize method
+      const options = {
+        attributes: ['id', 'year', 'subject', 'exam', 'mark', 'term', 'student_result_id'],
+        where: {},
+        include: [{model: Users, 'as': 'User'}],
```
Comment (jamsgra) [inline server/controllers/resultsController.js]: Should this be the same as `getAllResults`?

`include: [{model: Users, 'as': 'User', attributes: ['user_uid', 'first_name', 'last_name', 'dob', 'year_of_graduation', 'role', 'phone_number', 'email']}]`

Or if this is just retrieving the results of the logged in user, then maybe we don't need to return the `Users` table data at all.

What do you think?

Comment (sulenchy) [reply server/controllers/resultsController.js]: @jamsgra Just like you suggested, removing the `User` would be logical here.

**Pull request merged**